src/System.Windows.Forms/src/System/Windows/Forms/TrackBar.cs

```diff
@@ -557,11 +566,26 @@ public int TickFrequency
                 }
 
                 _tickFrequency = value;
-                if (IsHandleCreated)
+                // set the recreateHandle flag to null,
+                // if the _autoDrawTicks flag does not equal what ShouldAutoDrawTicks()
+                // we set the recreateHandle to true,
+                // then we skip sending the messages and,
+                // recreateHandle will be called at the end of the method.
```

Comment:
A good comment is the one explaining "why" or "how" rather than restating what the code does.
I would rephase it to something like the following:
```suggestion
                // Determine if the decision of whether the ticks drawing is performed by the native control or the Windows Forms runtime
                // is still valid. If it's no longer valid, we'll need to recreate the native control.
```

Also, this repo strives to wrap lines between 120-150 chars, so the text should be reflowed.